dictionary.go

```diff
@@ -181,7 +181,7 @@ func newFixedLenByteArrayDictionary(typ Type, bufferSize int) *fixedLenByteArray
 	return &fixedLenByteArrayDictionary{
 		typ:    typ,
 		size:   size,
-		values: make([]byte, 0, dictCap(bufferSize, size)),
+		values: make([]byte, 0, size),
```

Comment:
While it fixes the bug, it discard the buffer size hint; would this approach address the issue while retaining the use of `bufferSize`?
```suggestion
		values: make([]byte, 0, dictCap(bufferSize, size) * size),
```
`dictCap` returns a number of items, so if we scale it back to a number of bytes by multiplying by the size of an item it should address the issue?